semantics/src/namespace/scopes.rs

```diff
@@ -225,7 +222,13 @@ impl BlockScope {
         }
     }
 
-    pub fn add_var(&mut self, name: String, typ: Type) {
+    /// Retrieve the return type expected in the current function scope
+    pub fn return_type(&self) -> FixedSize {
```

Comment:
Maybe `function_return_type` or `func_return_type` would be slightly clearer. Sure block scopes don't have a return type otherwise (but they could) but I feel this would add a bit more clarity to the method.